src/interface_elements.jl

```diff
@@ -0,0 +1,449 @@
+    ip = cv.ip
+    nvhere, nvthere = get_n_dofs_on_side(vertexdof_indices, ip, :here), get_n_dofs_on_side(vertexdof_indices, ip, :there)
+    nfhere, nfthere = get_n_dofs_on_side(facedof_interior_indices, ip, :here), get_n_dofs_on_side(facedof_interior_indices, ip, :there)
+    nchere, ncthere = get_n_dofs_on_side(celldof_interior_indices, ip, :here), get_n_dofs_on_side(celldof_interior_indices, ip, :there)
+    if i ≤ nvhere
+        return :here, i
+    elseif i ≤ nvhere + nvthere
+        return :there, i - nvhere
+    elseif i ≤ nvhere + nvthere + nfhere
+        return :here, i - nvthere
+    elseif i ≤ nvhere + nvthere + nfhere + nfthere
+        return :there, i - nvhere - nfhere
+    elseif i ≤ nvhere + nvthere + nfhere + nfthere + nchere
+        return :here, i - nvthere - nfthere
+    elseif i ≤ nvhere + nvthere + nfhere + nfthere + nchere + ncthere
+        return :there, i - nvhere - nfhere - nchere
+    end
+    throw(ArgumentError("No baseindex for interface index $(i) for interpolation $(ip)."))
+end
+
+"""
+    get_base_value(get_value::Function, cv::InterfaceCellValues, qp::Int, i::Int, here::Bool)
+
+Return a value from an `::InterfaceCellValues` by specifing:
+- `get_value`: function specifing which kind of value, e.g. by passing `shape_value`
+- `qp`: index of the quadrature point
+- `i`: index of the base function
+- `here`: side of the interface, where `true` means "here" and `false` means "there".
+"""
+function get_base_value(get_value::Function, cv::InterfaceCellValues, qp::Int, i::Int, here::Bool)
+    side, baseindex = get_side_and_baseindex(cv, i)
+    if side == :here && here
+        return get_value(cv.here, qp, baseindex)
+    elseif side == :there && ! here
+        return get_value(cv.there, qp, baseindex)
+    else
+        return 0
+    end
+end
+
+"""
+    shape_value(cv::InterfaceCellValues, qp::Int, i::Int, here::Bool)
+
+Return the value of shape function `i` evaluated in quadrature point `qp`
+on side `here`, where `true` means "here" and `false` means "there".
+"""
+shape_value(cv::InterfaceCellValues, qp::Int, i::Int, here::Bool) = get_base_value(shape_value, cv, qp, i, here)
+
+"""
+    shape_gradient(cv::InterfaceCellValues, qp::Int, i::Int, here::Bool)
+
+Return the gradient of shape function `i` evaluated in quadrature point `qp`
+on side `here`, where `true` means "here" and `false` means "there".
+"""
+shape_gradient(cv::InterfaceCellValues, qp::Int, i::Int, here::Bool) = get_base_value(shape_gradient, cv, qp, i, here)
+
+"""
+    shape_value_average(cv::InterfaceCellValues, qp::Int, i::Int)
+
+Return the value of shape function `i` evaluated in quadrature point `qp`
+for computing the average value on an interface.
+"""
+function shape_value_average(cv::InterfaceCellValues, qp::Int, i::Int)
+    side, baseindex = get_side_and_baseindex(cv, i)
+    return side == :here ? 0.5*shape_value(cv.here, qp, baseindex) : 0.5*shape_value(cv.there, qp, baseindex)
```

Comment:
I think it would be better to divide by 2 since otherwise it will promote to `Float64` (since `0.5` is a `Float64`):
```suggestion
    return side == :here ? shape_value(cv.here, qp, baseindex) / 2 : shape_value(cv.there, qp, baseindex) / 2
```